UITests/NavigationTests.swift

```diff
@@ -159,19 +159,67 @@ class NavigationTests: KIFTestCase, UITextFieldDelegate {
         tester().longPressViewWithAccessibilityLabel("Reload", duration: 1)
         tester().waitForViewWithAccessibilityLabel("Request Mobile Site")
 
-        // Navigate to different URL
+        // Navigate to different URL on the same host
         tester().tapViewWithAccessibilityLabel("Cancel")
         tester().tapViewWithAccessibilityIdentifier("url")
         tester().clearTextFromAndThenEnterTextIntoCurrentFirstResponder("\(webRoot)/numberedPage.html?page=2\n")
         tester().waitForWebViewElementWithAccessibilityLabel("Page 2")
 
-        // The new navigation should load the mobile site with an offer to request the desktop site
+        // The new navigation (on the same host) should preserve the desktop site with an offer to request the mobile site
+        tester().longPressViewWithAccessibilityLabel("Reload", duration: 1)
+        tester().waitForViewWithAccessibilityLabel("Request Mobile Site")
+
+        // Navigate to different URL on a different host
+        tester().tapViewWithAccessibilityLabel("Cancel")
+        tester().tapViewWithAccessibilityIdentifier("url")
+        tester().clearTextFromAndThenEnterTextIntoCurrentFirstResponder("www.mozilla.org\n")
```

Comment:
Not sure if this is acceptable since it loads an external site. I couldn't find any better way to navigate to a different host, however.